### PR TITLE
Adding MacOS Support

### PR DESCRIPTION
**Pull request by njelich**
Hi! I played around with getting this to work on Mac, two main elements to get around were the linux_personality usage and lsb_release for getting some info. Both of those should be working fine, but I am getting some different behavior from tests.

I couldn't dig much deeper into these, but in general I get something like this in these tests, related to GDB

```
Stdout GDB output: ["", "$4 = void", "", "", "", ""]
Error: SiginfoParse("Siginfo string: $4 = void doesn't match regex template")
.
 Stderr: Error: Program terminated (no crash)
```

I would appreciate some direction. Thanks :) Hopefully we can port this amazing tool of Mac, and if LLDB is implemented instead of GDB even on M1 Macs (since GDB doesn't support M1 yet).

Note: the shift aways from `lsb_release` might be a good addition on its own, shifting aeverything to a single command - `uname`

### CODE REVIEW

**Comment by njelich** [comment]
Requesting reviewer from @anfedotoff

**Comment by SweetVishnya** [comment]
AFAIK, `uname` won't print Ubuntu distro version.

**Review by anfedotoff — CHANGES_REQUESTED** [review]

**Comment by Avgor46** [comment]
> I would appreciate some direction

We use precompiled binaries in tests, so they are in elf format. I think you should try to recompile them.

**Comment by anfedotoff** [comment]
Hi, @njelich!
Thanks, it's a great idea to support Mac!
As for GDB tests, the error occurs [here](https://github.com/anfedotoff/gdb-command/blob/master/src/siginfo.rs#L34). 
`$4 = void` looks like that no signal was cached by gdb. We use this [command](https://github.com/anfedotoff/gdb-command/blob/master/src/lib.rs#L316) for that.
I think, we should recompile gdb tests for Mac, to get them work. You can look at this [Makefile](casr/tests/casr_tests/x86_Makefile).
Maybe the right way is to compile binaries in tests, the way we do [here](https://github.com/ispras/casr/blob/master/casr/tests/tests.rs#L1730) for example.

**Comment by Avgor46** [comment]
I found that vanilla gdb from brew could not run properly on MacOS due to [this](https://sourceware.org/gdb/wiki/PermissionsDarwin)

File: .github/workflows/darwin-i386.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: darwin-i386
```
Comment (anfedotoff) [inline .github/workflows/darwin-i386.yml:1]: Let's rename darwin-i386 to darwin-x86-64. I suppose, macos-latest runs on x86-64, according to  https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md. Also we should rename .yaml file and job name too.

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Comment by SweetVishnya** [comment]
The code itself looks good to me. The next thing left is to recompile tests with Makefile.

**Comment by SweetVishnya** [comment]
Btw, the `Brew packages` CI step fails

File: libcasr/src/report.rs [hunk 2]
```diff
@@ -272,20 +272,44 @@ impl CrashReport {
...
+                        }
+                        _ => {}
+                    });
+                }
+            }
+        }
+        #[cfg(target_os = "macos")]
+        {
+            // Use sw_vers
+            let mut os_cmd = Command::new("sw_vers");
```
[the remark this answers is not in the document]
Comment (njelich) [reply libcasr/src/report.rs:297]: Well, `sw_vers` is a built in utility of the MacOS, so there is no need to install it.

**Comment by njelich** [comment]
@SweetVishnya I made progress with the CI and GDB install on M1 Mac. Re recompiling the tests, do I simply run the [x86 Makefile](https://github.com/ispras/casr/blob/master/casr/tests/casr_tests/x86_Makefile) at the start of the test CI?

**Comment by Avgor46** [comment]
We have the following tests:
 
1) coredump tests
2) clustering tests
3) casr-san tests
4) casr-gdb tests
5) other languages tests
6) fuzzers



The first kind should not be tested on Mac OS. Second works well. For 3-4 you just need to recompile using Makefile. Tests for 5-6 categories we should discuss.

**Comment by SweetVishnya** [comment]
However, there is still some ELF and signal dependent code:

https://github.com/ispras/casr/blob/master/libcasr/src/gdb/exploitable.rs

**Comment by njelich** [comment]
I disabled the coredump tests, it passes now

File: README.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 [![amd64](https://github.com/ispras/casr/actions/workflows/amd64.yml/badge.svg?branch=master)](https://github.com/ispras/casr/actions/workflows/amd64.yml)
 [![aarch64](https://github.com/ispras/casr/actions/workflows/aarch64.yml/badge.svg?branch=master)](https://github.com/ispras/casr/actions/workflows/aarch64.yml)
 [![riscv64](https://github.com/ispras/casr/actions/workflows/riscv64.yml/badge.svg?branch=master)](https://github.com/ispras/casr/actions/workflows/riscv64.yml)
+[![darwin](https://github.com/ispras/casr/actions/workflows/darwin.yml/badge.svg?branch=master)](https://github.com/ispras/casr/actions/workflows/darwin.yml)
```
Comment (SweetVishnya) [inline README.md:9]: Maybe, we should explicitly state that it is darwin-x86?

**Comment by njelich** [comment]
@SweetVishnya What should it be named then?

https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/
Because there will be no more Intel macs

**Comment by SweetVishnya** [comment]
@Avgor46 could you help with proper naming, please?

**Comment by Avgor46** [comment]
Maybe `darwin-arm64`?

**Comment by njelich** [comment]
Updated to `darwin-arm64`, please verify and merge.

**Review by SweetVishnya — APPROVED** [review]

**Comment by SweetVishnya** [comment]
@njelich, althoug, almost all of the tests are disabled. Have you tried to manually run casr-libfuzzer and casr-gdb? Do you have any screenshots?

**Pull request merged**